miner/worker.go

```diff
@@ -645,6 +645,67 @@ func (miner *Miner) fillTransactions(interrupt *atomic.Int32, env *environment)
 	prioPlainTxs, normalPlainTxs := make(map[common.Address][]*txpool.LazyTransaction), pendingPlainTxs
 	prioBlobTxs, normalBlobTxs := make(map[common.Address][]*txpool.LazyTransaction), pendingBlobTxs
 
+	// For X Layer
+	type okPayTx struct {
+		account common.Address
+		tx      *txpool.LazyTransaction
+	}
+
+	okPayTxs := make(map[common.Address][]*txpool.LazyTransaction)
+
+	sortedOkPayTxs := common.OrderedList[okPayTx]{}
+	sortedOkPayTxs.SetCompareFunc(func(a, b okPayTx) int {
+		if a.tx.Time.Before(b.tx.Time) {
+			return -1
+		}
+		return 1
+	})
```

Comment:
Should perhaps handle equality in the comparator
```suggestion
	sortedOkPayTxs.SetCompareFunc(func(a, b okPayTx) int {
	   if a.tx.Time.Before(b.tx.Time) {
            return -1
        }
        if a.tx.Time.After(b.tx.Time) {
            return 1
        }
        return 0
	})
```